### PR TITLE
Uploading/Handling image servlets and tests. 

### PR DESCRIPTION
**Pull request by lattai**
Created and tested servlets for making blobstores to allow players to upload pictures. I also made tests for the user authentication because I hadn't done that last time.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/ImageHandlerServlet.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    // Gets the URL of the uploaded file; null if no file was uploaded
+    private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+        BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+        Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+        List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+        // User submitted form without selecting a file
+        if (blobKeys == null || blobKeys.isEmpty()) {
+        return null;
```
Comment (lzoebaker) [inline src/main/java/com/google/sps/servlets/ImageHandlerServlet.java:103]: indent `return null`;

File: src/main/java/com/google/sps/servlets/ImageHandlerServlet.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        }
+
+        // Our form only contains a single file input, so get the first index.
+        BlobKey blobKey = blobKeys.get(0);
+
+        // User submitted form without selecting a file (live server)
+        BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+        if (blobInfo.getSize() == 0) {
+        blobstoreService.delete(blobKey);
```
Comment (lzoebaker) [inline src/main/java/com/google/sps/servlets/ImageHandlerServlet.java:112]: indent lines below `if` block

File: src/main/java/com/google/sps/servlets/ImageHandlerServlet.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        Key playerKey;
+
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()){
+            if (email.equals(entity.getProperty(EMAIL_PARAMETER).toString())){
+                playerKey = entity.getKey();
+            }
+        }
+        
+        return null;
```
Comment (lzoebaker) [inline src/main/java/com/google/sps/servlets/ImageHandlerServlet.java:163]: What happens when `null` is returned everywhere? Is a default image uploaded?

File: src/test/java/com/google/sps/UploadImageTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.mockito.MockitoAnnotations;
+import java.io.IOException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+// Responsible for testing the UploadImage and ImageHandler Servlets to make sure
```
Comment (lzoebaker) [inline src/test/java/com/google/sps/UploadImageTest.java:30]: Use javadoc formatting https://google.github.io/styleguide/javaguide.html#s7-javadoc

File: src/test/java/com/google/sps/UploadImageTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  public void uploadsProperly() throws IOException {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter  = new PrintWriter(stringWriter);
+    when(this.response.getWriter()).thenReturn(printWriter);
+        
+    this.uploadImageServlet.doGet(this.request, this.response);
+    String result = stringWriter.toString();
+    Assert.assertTrue(result.contains(EXPECTED_JSON_OUTPUT));
+  }
+
```
Comment (lzoebaker) [inline src/test/java/com/google/sps/UploadImageTest.java:64]: Add tests for behavior when null is returned/ user does not upload image.
